ui/shared/RevisionLinkify.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReactLinkify, { linkify } from 'react-linkify';
+
+const revRe = /([a-f\d]{12,40})/;
+
+export default class RevisionLinkify extends React.Component {
+  constructor(props) {
+    super(props);
+
+    linkify.add('rev:', {
+      validate: function (text, pos, self) {
```

Comment:
I must have accidentally deleted the original comment, so I'm putting it back so your fixes make sense :) 

I wrote:
I'm curious why you're using a function expression here rather than an arrow function as below?